lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2491,6 +2574,28 @@ static function ( $split_selector ) use ( $clean_style_variation_selector ) {
 			}
 		}
 
+		$style_variation_block_declarations = array();
+		if ( ! empty( $block_metadata['variation_nodes'] ) ) {
+			foreach ( $block_metadata['variation_nodes'] as $variation_node ) {
+				$style_variation_block_node   = _wp_array_get( $this->theme_json, $variation_node['path'], array() );
+				$variation_block_declarations = static::get_feature_declarations_for_node( $variation_node, $style_variation_block_node );
+
+				foreach ( $variation_block_declarations as $current_selector => $new_declarations ) {
+					$style_variation_block_declarations[ $current_selector ] = $new_declarations;
+				}
+
+				// Note that `get_feature_declarations_for_node` will also unset
+				// feature values so they aren't duplicated in declarations via
+				// the call below.
```

Comment:
I'm not sure I understand this comment. Which feature values are unset by `get_feature_declarations_for_node` and why might they be duplicated below? Might be worth adding a bit more context. (Also is it correct to assume "features" here means the feature selectors optionally declared by blocks?)